miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/userinfo/UserInfoBridgeDispatcher.kt

```diff
@@ -22,16 +22,46 @@ abstract class UserInfoBridgeDispatcher {
      * Get user name from host app.
      * You can also throw an [Exception] from this method to pass an error message to the mini app.
      */
+    @Deprecated(
+        "This function has been deprecated.",
+        ReplaceWith("getUserName(onSuccess: (name: String) -> Unit, onError: (message: String) -> Unit)")
+    )
     open fun getUserName(): String =
         throw MiniAppSdkException("The `UserInfoBridgeDispatcher.getUserName` $NO_IMPL")
 
+    /**
+     * Get user name from host app.
+     * You can also throw an [Exception] from this method to pass an error message to the mini app.
+     */
+    open fun getUserName(
+        onSuccess: (name: String) -> Unit,
+        onError: (message: String) -> Unit
```

Comment:
We can use a single callback with [Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/).